File: solutions/exercise-4/src/main/java/io/miragon/training/application/service/RevokeClaimService.java
package io.miragon.training.application.service;

import io.miragon.training.application.port.inbound.RevokeClaimUseCase;
import io.miragon.training.application.port.outbound.MembershipRepository;
import io.miragon.training.domain.MembershipId;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional
public class RevokeClaimService implements RevokeClaimUseCase {

    private static final Logger log = LoggerFactory.getLogger(RevokeClaimService.class);

    private final MembershipRepository repository;

    public RevokeClaimService(MembershipRepository repository) {
        this.repository = repository;
    }

    @Override
    public void revokeClaim(MembershipId membershipId) {
        var membership = repository.find(membershipId);
        log.info("Revoking membership claim for {} (membershipId={})", membership.email().value(), membershipId.value());
    }
}

File: solutions/exercise-6/src/main/kotlin/io/miragon/training/application/service/RevokeClaimService.kt
package io.miragon.training.application.service

import io.github.oshai.kotlinlogging.KotlinLogging
import io.miragon.training.application.port.inbound.RevokeClaimUseCase
import io.miragon.training.application.port.outbound.MembershipRepository
import io.miragon.training.domain.MembershipId
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional
class RevokeClaimService(
    private val repository: MembershipRepository,
) : RevokeClaimUseCase {

    private val log = KotlinLogging.logger {}

    override fun revokeClaim(membershipId: MembershipId) {
        val membership = repository.find(membershipId)
        log.info { "Revoking membership claim for ${membership.email.value} (membershipId=${membershipId.value})" }
    }
}
